src/ctr_logging.c

```diff
@@ -426,6 +432,20 @@ static int write_k8s_log(stdpipe_t pipe, const char *buf, ssize_t buflen)
 	char tsbuf[TSBUFLEN];
 	set_k8s_timestamp(tsbuf, sizeof tsbuf, stdpipe_name(pipe));
 
+	/* If buflen is 0, this is a drain operation. Generate terminating F-sequence if needed. */
+	if (buflen == 0 && *has_partial) {
+		/* Generate terminating F-sequence for previous partial line */
+		if (writev_buffer_append_segment(k8s_log_fd, &bufv, tsbuf, TSBUFLEN - 1) < 0) {
+			nwarn("failed to write timestamp for terminating F-sequence");
+		} else if (writev_buffer_append_segment(k8s_log_fd, &bufv, "F\n", 2) < 0) {
+			nwarn("failed to write terminating F-sequence");
+		} else {
+			k8s_bytes_written += TSBUFLEN - 1 + 2;
```

Comment:
hmm in the case where we succeed in the first branch but fail in the second, I don't think we'd get here but we did write TSBUFLEN - 1 to the buffer. I think your branching strategy is slick but maybe not completely accurate and it could be simplified to be a bit clearer what's going on. it took me a moment to understand

Reply:
I will change the logic to use bools, not if/else to be more transparent.